.github/workflows/pre-commit.yml

```diff
@@ -16,6 +16,9 @@ concurrency:
 jobs:
   pre-commit:
     runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        python-version: ["current", "next", "previous"]
```

Comment:
Should we make this `python-version: ["3.9", "3.10", "3.11"]` to be a little more explicit in what we support?

Reply:
It's like this for very good reason - has to do with `required_status_checks` and matrix-management. There's all sorts of issues around renaming checks/matrix members and required checks. When we'll bump python version we'll be able to do it centrally in `.github/actions/setup-backend/action.yml` (current/next/future is defined by us in there), and we won't have issues around having to rename check names in master prior to being able to merge the branch

Reply:
This recent PR shows the kind of trouble we have to go through in order to rename checks... https://github.com/apache/superset/pull/29866